apps/wallet-mobile/src/features/SetupWallet/useCases/RestoreWallet/RestoreWalletScreen.tsx

```diff
@@ -35,6 +48,64 @@ export const RestoreWalletScreen = () => {
     }, [mnemonicType, track]),
   )
 
+  const intl = useIntl()
+  const {navigateToTxHistory} = useWalletNavigation()
+  const selectWalletMeta = useSetSelectedWalletMeta()
+  const selectWallet = useSetSelectedWallet()
+
+  const {openWallet} = useOpenWallet({
+    onSuccess: ([wallet, walletMeta]) => {
+      selectWalletMeta(walletMeta)
+      selectWallet(wallet)
+      navigateToTxHistory()
+    },
+    onError: (error) => {
+      InteractionManager.runAfterInteractions(() => {
+        return error instanceof InvalidState
+          ? showErrorDialog(errorMessages.walletStateInvalid, intl)
+          : error instanceof NetworkError
+          ? showErrorDialog(errorMessages.networkError, intl)
+          : showErrorDialog(errorMessages.generalError, intl, {message: error.message})
+      })
+    },
+  })
+
+  const saveMnemonic = React.useCallback(async () => {
+    const {accountPubKeyHex} = await makeKeys({mnemonic})
+    const {accountPlate} = await generateShelleyPlateFromKey(accountPubKeyHex, 1, networkId)
```

Comment:
need to consider byron wallets

Reply:
what this hardcoded `1` represents?

Reply:
bare in mind that `networkId` is not the same of `chainId`, for the plate my guess is that 0/1 mainnet/testnet should be used and not the internal Yoroi network id. (this needs an refactor later) is super misleading.

Reply:
> what this hardcoded 1 represents?

address count

> bare in mind that networkId is not the same of chainId, for the plate my guess is that 0/1 mainnet/testnet should be used and not the internal Yoroi network id. (this needs an refactor later) is super misleading.

`networkId` is needed because it will be used to create an `AddressGenerator` instance, where `networkId` is required

Reply:
@banklesss I think you can call directly wallleChecksum from cip4-js package.

Reply:
done